File: 2579.py
n=int(input())
array=[0]*(n+1)
dp=[0]*(n+1)
for i in range(1,n+1):
    array[i]=int(input())

for i in range(1,n+1):
    if i==1:
        dp[1]=array[1]
    elif i==2:
        dp[2]=array[1]+array[2]
    elif i==3:
        dp[3]=max(array[1]+array[3], array[2]+array[3])
    else:
        dp[i]=max(dp[i-3]+array[i-1]+array[i], dp[i-2]+array[i])

print(dp[n])

File: 2579.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int N, x;
    cin>>N;
    int stairs[N+1];
    int dp[N+1];
    for (int i=0; i<N; i++){
        cin>>x;
        stairs[i+1]=x;
    }
    dp[0]=0;
    if (N==1){
        cout<<stairs[1]<<"\n";
    }
    else if (N==2){
        cout<<stairs[1]+stairs[2]<<"\n";
    }
    else{
        dp[1]=stairs[1];
        dp[2]=stairs[1]+stairs[2];
        for (int i=3; i<=N; i++){
            dp[i]=max(dp[i-2], stairs[i-1]+dp[i-3])+stairs[i];
        }
        cout<<dp[N]<<"\n";
    }
    return 0;
}